plasoscaffolder/plugins/interface.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+"""The plugin interface classes."""
+import abc
+import collections
+
+from typing import Tuple
+
+from plasoscaffolder.lib import definitions
+from plasoscaffolder.lib import errors
+
+
+Question = collections.namedtuple(
+    'question', ['attribute', 'prompt', 'help', 'type'])
+
+
+class ScaffolderPlugin(object):
+  """The plugin interface."""
+
+  # The name of the plugin or parser this scaffolder plugin provides.
+  NAME = 'base_parser'
+
+  # One liner describing what the plugin provides.
+  DESCRIPTION = ''
+
+  # Define which project this particular plugin belongs to.
+  PROJECT = definitions.PROJECT_UNDEFINED
+
+  # Questions, a list that contains all the needed questions that the
+  # user should be prompted about before the plugin or parser is created.
+  # Each element in the list should be of the named tuple question.
+  QUESTIONS = []
+
+  def __init__(self):
+    """Initializes the plugin."""
+    super(ScaffolderPlugin, self).__init__()
+    self._attributes = {}
+    self._defined_attributes = set()
+    self._output_name = ''
+
+  def GetQuestions(self) -> list:
+    """Returns all plugin questions."""
+    return self.QUESTIONS
+
+  @abc.abstractmethod
+  def GenerateFiles(self) -> Tuple[str, str]:
+    """Generate all the files this plugin provides.
+
+    Yields:
+      list: file name and content of the file to be written to disk.
+    """
+
+  @abc.abstractmethod
+  def GetFilesToCopy(self) -> Tuple[str, str]:
+    """Return a list of files that need to be copied.
+
+    Yields:
+      tuple (str, str): file name of source and destination.
+    """
+
+  def IsPluginConfigured(self):
```

Comment:
Rename to RaiseIfNotReady or similar - based on the name, I'd expect this to return a bool.

Reply:
done